tests/Signals_perftest.m

```diff
@@ -1,90 +1,191 @@
 classdef Signals_perftest < matlab.perftest.TestCase
+% Runs performance tests for Signals
+%
+% Performs benchmarking for various operations on signals: 
+% nops, assignment ops, logical ops, scalar arithmetic ops, listener ops,
+% and signal methods: `map`, `scan`, `identity`, `subscriptable`, etc...
+% (via `test_mainOps` and `test_signalsOps` methods)
+%
+% And performs benchmarking for propogations through each signal in 
+% networks of various widths and depths (30 to 1000 nodes spread across 2
+% to 20 layers) (via `test_networkOps` method) (*Note: for reference, the 
+% `docs\examples\advancedChoiceWorld` exp def has 338 nodes over 10 layers)
+%
+% In this class, the 'Net' property is a Signals network, in which the 
+% 'A' & 'B' properties are treated as "input layer" signals from which all 
+% other signals are defined.
+%   
+% Example - display the mean measured time for running each test:
+%   results = runperf('Signals_perftest2.m');
+%   fullTableResults = vertcat(results.Samples);
+%   meanTimeByTestTable = varfun(@mean, fullTableResults,...
+%     'InputVariables', 'MeasuredTime', 'GroupingVariables', 'Name')
   
   properties
-    net
-    A
-    B
-    C
-    func
-    verify = false
-    results
-    reps = 2000
+    Net % Signals network ('sig.Net' object)
+    A % origin signal ('sig.node.OriginSignal' object)
+    B % origin signal ('sig.node.OriginSignal' object)
+    Reps = 1000 % number of test repetitions 
   end
   
   properties (TestParameter)
-    ops = num2cell([1:6 10:14 30 40 50])
+    % main operations to test (whose results are posted in the paper)
+    MainOps = {@nop, @post, @gt, @ge, @lt, @le, @eq,... 
```

Comment:
Can we please return to the previous distinction of MEX ops vs others?  I had created a containers.Map object so that one could test the performance of a specific op code.  This is useful because if we add new op codes in the future they can be added to the performance test in a more informative way and perhaps compared with a higher lever implementation of the same function. 

The new distinction between 'mainOps', 'basicOps' and 'SignalsOps' appears to be arbitrarily based on what was put in the paper(?)

I'd prefer it if we had two tests: one tests the MEX ops (the test_ops function I created, which takes an op code as its input), and another that tests Signals methods that don't directly map to an op code, including testing `map(@identity)` and `scan`.

Reply:
The mex ops are segeregated in 'BasicOps'.  'SignalsOps' is for signals methods that don't map to an op code. 'MainOps' is for the paper and for future benchmark comparison. The 'containers.Map' variable seemed unnecessary to me - for adding future functions, you can just add the handle directly to the parameter array. It helps visibility this way too. How do you see using the 'containers.Map' object as being more informative?